### PR TITLE
[WIP] Update to React 16, added prettier

### PR DESCRIPTION
**Pull request by aulneau**
This PR updates some deps throughout the project, most notably react v15 -> react 16. This should give us some perf improvements.

I've also added [Prettier](https://prettier.io/) to this repo. Prettier is a tool similar to eslint but it's main goal is to help projects maintain code consistency. It's absolutely fantastic and has become a real standard in the js community today. I've added a task you can run before committing, `yarn/npm run format` and it will run prettier throughout the codebase.

This PR doesn't change any code, just updates deps and added some formatting functionality. I did have to change all the imports of `PropTypes` from `'react`' to the new separate `prop-types` package.

Also I noticed in the `package.json` the task for building the production version is named `prod`. Typically the `build` task is reserved for that. Services like heroku and now.sh look for that task to automatically deploy and build the application. Would there be any hesitation in switching from the task name `prod` -> `build`?

I have built the app for production and deployed a version of it here -> https://blockstack-staging.ineffable.digital/

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "useTabs": false,
+  "arrowParens": "always",
+  "printWidth": 80,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "trailingComma": "all",
+  "jsxBracketSameLine": false,
+  "semi": true
```
Comment (cwackerfuss) [inline .prettierrc]: I think `semi` should be set to false. I prefer to leave them out and it looks like the standard has been no semicolons in this project so far. @yknl @larrysalibra thoughts?

File: package.json [hunk 2]
```diff
@@ -79,13 +19,14 @@
     "lodash": "^3.10.1",
     "marked": "^0.3.6",
     "popper.js": "^1.12.5",
-    "prop-types": "^15.5.8",
+    "prettier": "^1.11.1",
```
Comment (cwackerfuss) [inline package.json]: Prettier should be in `devDependencies` not `dependencies`

File: package.json [hunk 3]
```diff
@@ -112,5 +54,64 @@
...
+  "scripts": {
+    "build-docs": "npm run fetch-docs; gulp buildDocs",
+    "dev": "npm run fetch-docs; gulp dev",
+    "start": "serve ./build --single",
+    "fetch-docs": "git submodule init; git submodule update --remote",
+    "postinstall": "cd ./node_modules/highlight.js && npm i && node ./tools/build.js :common",
+    "build": "npm run fetch-docs; gulp prod",
+    "prod": "npm run fetch-docs; gulp prod",
+    "format": "prettier --write \"app/**/*.js\" && echo -e '\\033[0;32m'💅 Just use prettier'\\033[0m'"
   }
```
Comment (cwackerfuss) [inline package.json]: Why'd you reorganize the package.json file (or was it prettier?)? The standard is scripts near the top before dependencies. Can we change that back? If we wanna reorganize we can always make another PR to discuss.

File: package.json (hunk 2 above)
Comment (aulneau) [reply package.json]: Fixed!

File: .prettierrc (hunk 1 above)
Comment (aulneau) [reply .prettierrc]: Sure np, I prefer semis but nbd!

File: package.json (hunk 3 above)
Comment (aulneau) [reply package.json]: For every new project I work on I run this on it: https://github.com/henrikjoreteg/fixpack

It organizes them in a consistent and predictable way, and if I ever modify a `package.json `manually, I like to run it to keep things tidy.

```
It will re-write your package.json file as follows:

name first
description second
version third
author fourth
all other keys in alphabetical order
dependencies and devDependencies sorted alphabetically
newline at the end of the file

It will warn you if any of these are missing:

description
author
repository
keywords
main
bugs
homepage
license
```

I don't know that this really warrants a new, separate PR, if there is a rational or previously agreed upon organizational structure of `package.json` files for blockstack, let me know and I'll switch it back!
